src/main/java/ai/chalk/internal/arrow/FeatherProcessor.java

```diff
@@ -28,8 +28,10 @@
 import java.util.*;
 
 public class FeatherProcessor {
-    public static final long ROOT_ALLOCATOR_SIZE = Long.MAX_VALUE;
-    public static final long CHILD_ALLOCATOR_SIZE = 1024 * 1024 * 1024;
+    public static final long ALLOCATOR_SIZE_REQUEST = 1_000_000_000;
+    public static final long ALLOCATOR_SIZE_RESPONSE = 10 * ALLOCATOR_SIZE_REQUEST;
+    public static final long ALLOCATOR_SIZE_ROOT = ALLOCATOR_SIZE_RESPONSE * 500;
```

Comment:
only concern here would be if they have more then 500 outstanding responses per client, likely not an issue because I suspect response size is actually <<< the limit